File: sources/java/09_fib_recursive/Main.java
// task 09 fib_recursive — expected output: 102334155
// build: javac Main.java    run: java Main
// build (graalvm native-image): native-image -O2 Main    run: ./Main
// Naive double recursion, no memoization.

public class Main {
    private static long fib(long n) {
        if (n < 2) {
            return n;
        }
        return fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        System.out.println(fib(40));
    }
}

File: sources/perl/09_fib_recursive/main.pl
# task 09 fib_recursive — expected output: 102334155
# build: perl main.pl    run: perl main.pl
# note: naive double recursion, no memoization; fib(40) is about 1.6 billion calls and perl is slow
#       enough that this may hit the 300 s benchmark timeout. That is a result, not a bug.
use strict;
use warnings;

sub fib {
    my ($n) = @_;
    return $n if $n < 2;
    return fib($n - 1) + fib($n - 2);
}

print fib(40), "\n";
